PluginGenerator/PluginBuilder.cs

```diff
@@ -152,17 +152,21 @@ public PluginBuilder AddReferencedJar(string fullJarFilePath)
             return this;
         }
 
-        public PluginBuilder AddExtension(string fullExtensionClassName)
+        /// <summary>
+        /// Registers an extension that will be exported to SonarQube
+        /// </summary>
+        /// <param name="extension"></param>
+        /// <remarks>The extension could be an instance of a class (such as a PropertyDefinition), or it could
+        /// be a type that SonarQube should instantiate (e.g. a subclass of RulesDefinition). If the extension is
+        /// a type then it should end with ".class".</remarks>
+        public PluginBuilder AddExtension(string extension)
         {
-            if (string.IsNullOrWhiteSpace(fullExtensionClassName))
```

Comment:
Previously we assumed that the extension would be a class (type). However, it could also be an instance of an object (e.g. a PropertyDefinition).